lib/ansible/galaxy/collection.py

```diff
@@ -479,6 +541,62 @@ def validate_collection_path(collection_path):
     return collection_path
 
 
+def verify_collections(collections, search_paths, apis, validate_certs, ignore_errors):
+
+    with _display_progress():
+        with _tempdir() as b_temp_path:
+            for collection in collections:
+                try:
+
+                    local_collection = None
+                    b_temp_tar_path = None
+                    b_collection = to_bytes(collection[0], errors='surrogate_or_strict')
```

Comment:
The collection name needs to be verified, otherwise there will be a stack trace on line 565.
```
> ansible-galaxy collection verify bad -vvv
ansible-galaxy 2.10.0.dev0
  config file = /Users/sdoran/.ansible.cfg
  configured module search path = ['/usr/local/share/ansible/library']
  ansible python module location = /Users/sdoran/Source/ansible/lib/ansible
  executable location = /Users/sdoran/Source/ansible/bin/ansible-galaxy
  python version = 3.7.4 (default, Oct 15 2019, 10:30:33) [Clang 11.0.0 (clang-1100.0.33.8)]
Using /Users/sdoran/.ansible.cfg as config file
ERROR! Unexpected Exception, this is probably a bug: not enough values to unpack (expected 2, got 1)
the full traceback was:

Traceback (most recent call last):
  File "/Users/sdoran/Source/ansible/bin/ansible-galaxy", line 123, in <module>
    exit_code = cli.run()
  File "/Users/sdoran/Source/ansible/lib/ansible/cli/galaxy.py", line 398, in run
    context.CLIARGS['func']()
  File "/Users/sdoran/Source/ansible/lib/ansible/cli/galaxy.py", line 841, in execute_verify
    verify_collections(requirements, resolved_paths, self.api_servers, (not ignore_certs), ignore_errors)
  File "/Users/sdoran/Source/ansible/lib/ansible/galaxy/collection.py", line 565, in verify_collections
    remote_collection = CollectionRequirement.from_name(collection[0], apis, collection[1], False, parent=None)
  File "/Users/sdoran/Source/ansible/lib/ansible/galaxy/collection.py", line 371, in from_name
    namespace, name = collection.split('.', 1)
ValueError: not enough values to unpack (expected 2, got 1)
```

Something like this should do the trick.
```suggestion
                    verify_collection_name(collection[0])
                    b_collection = to_bytes(collection[0], errors='surrogate_or_strict')
```